src/dmd/backend/type.d

```diff
@@ -79,6 +79,7 @@ struct TYPE
 
     uint Tcount; // # pointing to this type
     TYPE* Tnext; // next in list
+    TYPE* Tkey;  // typtr: key type for associative arrays
```

Comment:
I think you should move Tident out of the union because it is the commonly used field. It is also (ab)used by the dynamic array types and might also be used by the next type (delegates?). Adding usage information in the comments would be best.

Reply:
So what should I change? Move `Tident` from union as well? Or move `Tkey` back and only move `Tident` out of that union?

Reply:
I think Tident should be moved out of the union instead of Tkey. Using a union is meant to save memory reusing the same space for different types. Tkey fits in perfectly, while Tident is now used for multiple types.

Reply:
Moved `Tident` instead of `Tkey` as you suggested.